### ISSUE TITLE
delete Library Directory maybe cause NSUserDefault damage

### ISSUE THREAD

**Issue by xiaoshaogg** [liuslevis/GYBootingProtection#8]
delete Library Directory maybe cause NSUserDefault damage

**Comment by luoqisheng** [comment]
any update?

**Comment by liuslevis** [comment]
Thanks for report.
Would you provide any detail information or demo project about this?

**Comment by luoqisheng** [comment]
i'm using SwiftyUserDefaults, and after deleting all dirs, NSUserDefault can not work

**Comment by liuslevis** [comment]
You are free to customize what to delete.

Making it not to delete library dir may solve your problem.

**Comment by luoqisheng** [comment]
Yes, but userdefault was  the most widely used .  is there any other solution to clear all dirty data cached

**Comment by liuslevis** [comment]
Guess SwiftyUserDefaults rely on some folders under Library Directory.
You need to find out what folder it is, and avoid to delete these folder.

**Comment by luoqisheng** [comment]
The problem still exists even when we use NSUserDefaults
